common/stacks/src/main/scala/weco/api/stacks/services/SierraService.scala

```diff
@@ -34,15 +34,18 @@ class SierraService(
 )(implicit ec: ExecutionContext)
     extends Logging {
 
-  def getAccessCondition(
-    sourceIdentifier: SourceIdentifier
-  ): Future[Either[SierraItemLookupError, Option[AccessCondition]]] = {
-    val itemNumber = SierraItemIdentifier.fromSourceIdentifier(sourceIdentifier)
-
+  def getAccessConditions(
+    itemNumbers: Seq[SierraItemNumber]
+  ): Future[Either[SierraItemLookupError, Map[SierraItemNumber, Option[
+    AccessCondition
+  ]]]] = {
     for {
-      itemEither <- sierraSource.lookupItem(itemNumber)
-      accessCondition = itemEither.map { item =>
-        item.getAccessCondition(itemNumber)
+      itemEither <- sierraSource.lookupItemEntries(itemNumbers)
+      accessCondition = itemEither.map {
+        _.entries
+          .map(
+            item => item.id -> item.getAccessCondition
+        ) toMap
```

Comment:
Now looking at this function, hmm.

You're preserving the `SierraItemLookupError` here, and I'm not sure that's correct – if we know about some of the items, is it better to do a partial update or hard fail? I think a partial update is better, but that's difficult to communicate – without propagating the error, how do we distinguish between "we couldn't get updated item data" and "we got the updated item data, and there's now no access condition".

Which made me wonder… is there ever a case in which we won't return an access condition?

Spoiler: no. https://github.com/wellcomecollection/catalogue-pipeline/pull/1778

If you pull in that patch, you instantly lose a layer of `Option[…]` here – and I think it enables you to tidy up this signature further, and do something like:

```scala
def getUpdatedAccessConditions(
  itemNumbers: Seq[SierraItemNumber]
): Future[Map[SierraItemNumber, AccessCondition]]
```

Inside this function you log a warning about any missing items, but you don't let the error propagate any further (at least, not directly).

You return a Map of access conditions that should replace the conditions on the existing items. If an item number is missing from this Map, then the caller should keep whatever condition was already there from the Catalogue API.